src/Nethermind/Nethermind.Crypto/KzgPolynomialCommitments.cs

```diff
@@ -51,16 +73,29 @@ public static bool TryComputeCommitmentV1(ReadOnlySpan<byte> commitment, Span<by
         return false;
     }
 
-    public static unsafe bool VerifyProof(ReadOnlySpan<byte> commitment, ReadOnlySpan<byte> z, ReadOnlySpan<byte> y, ReadOnlySpan<byte> proof)
+    public static bool VerifyProof(ReadOnlySpan<byte> commitment, ReadOnlySpan<byte> z, ReadOnlySpan<byte> y,
+        ReadOnlySpan<byte> proof)
     {
-        if (_ckzgSetup == IntPtr.Zero)
+        try
+        {
+            return Ckzg.Ckzg.VerifyKzgProof(commitment, z, y, proof, _ckzgSetup);
+        }
+        catch (Exception e) when (e is ArgumentException or ApplicationException or InsufficientMemoryException)
         {
-            throw new InvalidOperationException("KZG is not initialized");
+            return false;
```

Comment:
Does it mean that we will mark tx as invalid and as a result the entire block as invalid in case of exception? Let's discuss it carefully

Reply:
This validation applies for pool tx only, so we can have a tx declined in a very rare and extreme case

Reply:
Just to make you aware: TxValidator is used also to check block validity in NewPayload and during the sync. So definitely, it is not TxPool only thing